signature-specification.md

```diff
@@ -120,125 +120,105 @@ Notary v2 requires the signature envelope to support the following signed attrib
-Notary v2 uses JWS JSON Serialization for the signature envelope with some additional constraints on the structure of claims and headers.
+In COSE ([rfc8152](https://datatracker.ietf.org/doc/html/rfc8152)), data is stored as either payload or headers (protected and unprotected).
+Notary v2 uses [COSE_Sign1_Tagged](https://datatracker.ietf.org/doc/html/rfc8152#section-4.2) object as the signature envelope with some additional constraints on the header fields.
 
 Unless explicitly specified as OPTIONAL, all fields are required.
-Also, there shouldn’t be any additional fields other than ones specified in JWSPayload, ProtectedHeader, and UnprotectedHeader.
 
-**JWSPayload a.k.a. Claims**:
-Notary v2 is using one private claim (`notary`) and two public claims (`iat` and `exp`).
-An example of the claim is described below
+**Payload**: COSE signs the payload as defined in the [Payload](#payload) section.
 
-```json
+**ProtectedHeaders**: Notary v2 supports only the following protected headers:
+
+```
 {
-   "subject": {
-      "mediaType": "application/vnd.oci.image.manifest.v1+json",
-      "digest": "sha256:73c803930ea3ba1e54bc25c2bdc53edd0284c62ed651fe7b00369da519a3c333",
-      "size": 16724,
-      "annotations": {
-         "key1": "value1",
-         "key2": "value2",
-         ...
-      }
-   },
-   "iat": 1234567891000,
-   "exp": 1234567891011
+  / crit / 2: [
+    / cty / 3,
+    'signingtime',
+    'expiry'
+  ],
+  / cty / 3: 'application/vnd.cncf.oras.artifact.descriptor.v1+json',
+  'signingtime': 1234567891000,
+  'expiry': 1234567891011
 }
 ```
 
-The payload contains the subject manifest and other attributes that have to be integrity protected.
+Note: The above example is represented using the [extended CBOR diagnostic notation](https://datatracker.ietf.org/doc/html/rfc8152#appendix-C).
 
-- **`subject`**(*descriptor*): A REQUIRED top-level node consisting of the manifest that needs to be integrity protected.
-  Please refer [Payload](#payload) section for more details.
-- **`iat`**(*number*): The REQUIRED property Issued-at(`iat`) identifies the time at which the signature was issued.
-- **`exp`**(*number*): This OPTIONAL property contains the expiration time on or after which the signature must not be considered valid.
+- **`crit`** (*array of integers or strings*): This REQUIRED property (label: `2`) lists the headers that implementation MUST understand and process.
+  The array MUST contain `3` (`cty`), and `signingtime`. If `expiry` is presented, the array MUST also contain `expiry`.
+- **`cty`** (*string*): The REQUIRED property content-type (label: `3`) is used to declare the media type of the secured content (the payload).
+- **`signingtime`** (*integer*): The REQUIRED property identifies the time at which the signature was generated.
+- **`expiry`** (*integer*): This OPTIONAL property contains the expiration time on or after which the signature must not be considered valid.
```

Comment:
Clarify that these are not just "integers", but tagged [CBOR date/time values](https://datatracker.ietf.org/doc/html/rfc7049#section-2.4.1). Consider supporting other time formats such as RFC3339 or future [CBOR time](https://datatracker.ietf.org/doc/html/draft-ietf-cbor-time-tag).